numpy/typing/tests/data/reveal/fromnumeric.pyi

```diff
@@ -103,17 +103,23 @@ reveal_type(np.ravel(c))  # E: ndarray[Any, Any]
 reveal_type(np.ravel(A))  # E: ndarray[Any, Any]
 reveal_type(np.ravel(B))  # E: ndarray[Any, Any]
 
-reveal_type(np.nonzero(a))  # E: tuple[ndarray[Any, Any]]
-reveal_type(np.nonzero(b))  # E: tuple[ndarray[Any, Any]]
-reveal_type(np.nonzero(c))  # E: tuple[ndarray[Any, Any]]
-reveal_type(np.nonzero(A))  # E: tuple[ndarray[Any, Any]]
-reveal_type(np.nonzero(B))  # E: tuple[ndarray[Any, Any]]
-
-reveal_type(np.shape(a))  # E: tuple[builtins.int]
-reveal_type(np.shape(b))  # E: tuple[builtins.int]
-reveal_type(np.shape(c))  # E: tuple[builtins.int]
-reveal_type(np.shape(A))  # E: tuple[builtins.int]
-reveal_type(np.shape(B))  # E: tuple[builtins.int]
+reveal_type(np.ravel(b))  # E: ndarray[Any, dtype[bool_]]
+reveal_type(np.ravel(f4))  # E: ndarray[Any, dtype[{float32}]]
+reveal_type(np.ravel(f))  # E: ndarray[Any, dtype[Any]]
+reveal_type(np.ravel(AR_b))  # E: ndarray[Any, dtype[bool_]]
+reveal_type(np.ravel(AR_f4))  # E: ndarray[Any, dtype[{float32}]]
+
+reveal_type(np.nonzero(b))  # E: tuple[ndarray[Any, dtype[{intp}]], ...]
+reveal_type(np.nonzero(f4))  # E: tuple[ndarray[Any, dtype[{intp}]], ...]
+reveal_type(np.nonzero(f))  # E: tuple[ndarray[Any, dtype[{intp}]], ...]
+reveal_type(np.nonzero(AR_b))  # E: tuple[ndarray[Any, dtype[{intp}]], ...]
+reveal_type(np.nonzero(AR_f4))  # E: tuple[ndarray[Any, dtype[{intp}]], ...]
+
+reveal_type(np.shape(b))  # E: tuple[builtins.int, ...]
+reveal_type(np.shape(f4))  # E: tuple[builtins.int, ...]
+reveal_type(np.shape(f))  # E: tuple[builtins.int, ...]
+reveal_type(np.shape(AR_b))  # E: tuple[builtins.int, ...]
+reveal_type(np.shape(AR_f4))  # E: tuple[builtins.int, ...]
```

Comment:
```suggestion
reveal_type(np.nonzero(a))  # E: tuple[ndarray[Any, Any], ...]
reveal_type(np.nonzero(b))  # E: tuple[ndarray[Any, Any], ...]
reveal_type(np.nonzero(c))  # E: tuple[ndarray[Any, Any], ...]
reveal_type(np.nonzero(A))  # E: tuple[ndarray[Any, Any], ...]
reveal_type(np.nonzero(B))  # E: tuple[ndarray[Any, Any], ...]

reveal_type(np.shape(a))  # E: tuple[builtins.int, ...]
reveal_type(np.shape(b))  # E: tuple[builtins.int, ...]
reveal_type(np.shape(c))  # E: tuple[builtins.int, ...]
reveal_type(np.shape(A))  # E: tuple[builtins.int, ...]
reveal_type(np.shape(B))  # E: tuple[builtins.int, ...]
```
It didn't backport quite as cleanly as I'd hoped, but this should resolve the issues.